Update cssparser to 0.11

SimonSapin opened the pull request:
<!-- Please describe your changes on the following line: -->

<s>Depends on https://github.com/servo/rust-cssparser/pull/122.</s>

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

<!-- Either: -->
- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/15735)
<!-- Reviewable:end -->

SimonSapin committed:
Update to cssparser 0.11

highfive commented:
Heads up! This PR modifies the following files:
 * @bholley: components/style/values/specified/basic_shape.rs, components/style/properties/longhand/text.mako.rs, components/style/values/specified/length.rs, components/style/properties/properties.mako.rs, components/style/gecko/media_queries.rs, components/style/Cargo.toml, components/style/lib.rs, components/style/values/specified/image.rs, components/style/properties/longhand/font.mako.rs, components/style/servo/media_queries.rs, components/style/properties/longhand/effects.mako.rs, components/style/values/specified/mod.rs, components/style/properties/longhand/counters.mako.rs, components/style/properties/longhand/box.mako.rs, components/style/supports.rs, components/style/values/mod.rs, components/style/values/specified/align.rs, components/style/properties/helpers/animated_properties.mako.rs, components/style/values/specified/position.rs, components/style/properties/longhand/inherited_svg.mako.rs
 * @kichjang: components/script/dom/csskeyframesrule.rs, components/script/lib.rs, components/script/Cargo.toml
 * @fitzgen: components/script/dom/csskeyframesrule.rs, components/script/lib.rs, components/script/Cargo.toml
 * @emilio: components/style/values/specified/basic_shape.rs, components/style/properties/longhand/text.mako.rs, components/style/values/specified/length.rs, components/style/properties/properties.mako.rs, components/style/gecko/media_queries.rs, components/style/Cargo.toml, components/style/lib.rs, components/style/values/specified/image.rs, components/style/properties/longhand/font.mako.rs, components/style/servo/media_queries.rs, components/style/properties/longhand/effects.mako.rs, components/style/values/specified/mod.rs, components/style/properties/longhand/counters.mako.rs, components/style/properties/longhand/box.mako.rs, components/style/supports.rs, components/style/values/mod.rs, components/style/values/specified/align.rs, components/style/properties/helpers/animated_properties.mako.rs, components/style/values/specified/position.rs, components/style/properties/longhand/inherited_svg.mako.rs

highfive commented:
<img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20> **Warning** <img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20>

* These commits modify style and script code, but no tests are modified. Please consider adding a test!

SimonSapin commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit 84aa188 with merge 1c993ff...

SimonSapin committed:
Use ascii_case_insensitive_phf_map! in PropertyId::parse

SimonSapin commented:
This is ready to land. r? @Manishearth

emilio commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit eacc12d has been approved by `emilio`

<!-- @bors-servo r=emilio eacc12d1e825a29d1dbe23a12a94c993da94abc8 -->

bors-servo commented:
:hourglass: Testing commit eacc12d with merge 11f362a...

bors-servo commented:
:broken_heart: Test failed - [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/2617)

SimonSapin commented:
```
{"status": "FAIL", "stack": null, "subtest": null, "action": "test_result", "test": "/_mozilla/css/overflow_wrap_a.html", "line": 64827, "message": "/_mozilla/css/overflow_wrap_a.html 57a783cf615a456badeb0f6c306887f9213b1ea0\n/_mozilla/css/overflow_wrap_ref.html 739aa7187d267c9ed72935ed31382631170de234\nTesting 57a783cf615a456badeb0f6c306887f9213b1ea0 == 739aa7187d267c9ed72935ed31382631170de234", "expected": "PASS"}
```

@bors-servo retry

bors-servo commented:
:hourglass: Testing commit eacc12d with merge b7109ad...

bors-servo commented:
:broken_heart: Test failed - [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/2619)

SimonSapin commented:
@bors-servo retry

bors-servo commented:
:zap: Previous build results for [android](http://build.servo.org/builders/android/builds/5535), [arm32](http://build.servo.org/builders/arm32/builds/5758), [arm64](http://build.servo.org/builders/arm64/builds/5783), [linux-dev](http://build.servo.org/builders/linux-dev/builds/5898), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/5893), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/2569), [windows-gnu-dev](http://build.servo.org/builders/windows-gnu-dev/builds/1055), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/1055) are reusable. Rebuilding only [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/2598), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/2619), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/5920), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/2641)...

SimonSapin commented:
#15738

bors-servo commented:
:broken_heart: Test failed - [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/2620)

SimonSapin committed:
Define keyword values in ASCII lowercase

The match_ignore_ascii_case! macro does ASCII-case-insensitive matching.
In cssparser verion 0.11, it will require its patterns to be already
lower case.

SimonSapin committed:
Replace 1-variant match_ignore_ascii_case! with expect_ident_matching

SimonSapin committed:
Borrow input to match_ignore_ascii_case!

In cssparser version 0.11, this macro will stop implicitly borrowing its
own input.

SimonSapin commented:
I thought the failure above was intermittent because I couldn’t reproduce locally at first, but it looks like this is a real bug. I must have messed up my initial testing. The bug was in the "Use `ascii_case_insensitive_phf_map!` in `PropertyId::parse`" commit: I forgot to add property aliases to the map. The test was using the `word-wrap` property, an alias for `overflow-wrap`. With that fixed, the test passes on my machine.

@bors-servo r=emilio

bors-servo commented:
:pushpin: Commit 2f996e3 has been approved by `emilio`

<!-- @bors-servo r=emilio 2f996e34f54cc2903d44945345bb0cd7f80922a3 -->

bors-servo commented:
:hourglass: Testing commit 2f996e3 with merge 94e563e...

bors-servo commented:
:bulb: This pull request was already approved, no need to approve it again.

- This pull request is currently being tested. If there's no response from the continuous integration service, you may use `retry` to trigger a build again.

bors-servo commented:
:pushpin: Commit 2f996e3 has been approved by `emilio`

<!-- @bors-servo r=emilio 2f996e34f54cc2903d44945345bb0cd7f80922a3 -->

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/5538), [arm32](http://build.servo.org/builders/arm32/builds/5761), [arm64](http://build.servo.org/builders/arm64/builds/5786), [linux-dev](http://build.servo.org/builders/linux-dev/builds/5901), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/2603), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/2623), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/5896), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/5924), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/2572), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/2645), [windows-gnu-dev](http://build.servo.org/builders/windows-gnu-dev/builds/1058), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/1058)
Approved by: emilio
Pushing 94e563e4d9292d7b19ce061e070cda358e822172 to master...
